assets/app/scripts/services/logs/logLinks.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('logLinks', [
+    '$anchorScroll',
+    '$document',
+    '$location',
+    '$window',
+    function($anchorScroll, $document, $location, $window) {
+      var doc = $document[0];
+      return {
+        makeDownload: function(obj) {
+          var a = doc.createElement('a');
+          a.href = $window.URL.createObjectURL(new Blob([obj], {type: 'text/plain'}));
+          a.download = 'log.txt';
+          doc.body.appendChild(a);
+          a.click();
+          doc.body.removeChild(a);
+        },
```

Comment:
Yeah, I'm inclined to `URL.revokeObjectURL(objectURL)` immediately after the `doc.body.removeChild(a);`.  The user will already have the file & we don't have to worry about memory.  I can't see a user clicking the link enough times to cause a performance issue, even if they ask for the same log more than once.  Its probably the cleanest way to ensure nothing lingers.